astronomy-pipeline/extract_functions.py

```diff
@@ -0,0 +1,185 @@
+"""File to obtain body positional data from the Astronomy API."""
+
+# pylint: disable=E0401,R0913,R0917
+
+from os import environ as ENV
+from datetime import date, timedelta
+import base64
+
+
+from psycopg2 import connect, extensions, extras
+from dotenv import load_dotenv
+import requests
+
+load_dotenv()
+
+ASTRO_URL = "https://api.astronomyapi.com/api/v2"
+
+
+def get_db_connection() -> extensions.connection:
+    """Reusable function for getting a database connection."""
+
+    return connect(cursor_factory=extras.RealDictCursor,
+                   dbname=ENV["DB_NAME"],
+                   host=ENV["DB_HOST"],
+                   password=ENV["DB_PASSWORD"],
+                   user=ENV["DB_USER"],
+                   port=ENV["DB_PORT"])
+
+
+def get_auth_string() -> str:
+    """Generates AstronomyAPI authorization key from env values."""
+
+    user_pass = f"{ENV["ASTRONOMY_ID"]}:{ENV["ASTRONOMY_SECRET"]}"
+
+    auth_string = base64.b64encode(user_pass.encode()).decode()
+
+    return auth_string
```

Comment:
Interesting security considerations

Reply:
This is just how the Astronomy API credentials need to be used to make authenticated requests.

Reply:
*authorisation 

We are not savages.

Reply:
Completely valid, please, take all of my gold stars, I am not worthy.

Has been updated.